### DISCUSSION TITLE
Will data be downloaded automatically using POST /api/ids/contract?

### DISCUSSION THREAD

**Discussion by jfernandezsqs** [International-Data-Spaces-Association/DataspaceConnector#109] [Q&A] [answered]
I am using DSC v6.2.0 and have ConnectorA acting as provider and connectorB acting as consumer.
I am having the following issue with the call POST /api/ids/contract
![image](https://user-images.githubusercontent.com/82888094/144445823-001ee361-5544-4e4c-96c9-d837ef8eabb7.png)
![image](https://user-images.githubusercontent.com/82888094/144445894-bb4199b5-593b-4a07-8314-6ee4043c9122.png)
I have specified that the connector should download automatically the data from the artifact but when I execute it there is no data automatically downloaded.
This issue also happens with the call GET /api/artifacts/{id}/data/**
![image](https://user-images.githubusercontent.com/82888094/144446093-a2414de0-8788-4e66-93c0-44a3e3a017c8.png)
![image](https://user-images.githubusercontent.com/82888094/144446149-da8c6b38-7184-4d57-91cb-3a00d06f9dfd.png)

Is this the expected behavior or is it a problem that the DSC is not downloading the data from the artifact?
Thanks in advance,

**Comment by juliapampus** [comment, marked as answer]
I think there is a misunderstanding. The Swagger UI will not automatically start downloading files to your download directory. The `download` boolean indicates whether the data is downloaded from the provider on request (via IDS `ArtifactRequestMessage`), or whether the consumer DSC loads the data from its internal database (if already donwloaded data exists). With this, you can also force the DSC to request the data from the provider again. Without this, no IDS connection is built up.
